lib/elixir/lib/map.ex

```diff
@@ -211,13 +220,16 @@ defmodule Map do
   """
   @spec take(map, [key]) :: map
   def take(map, keys) do
-    Enum.reduce(keys, [], fn key, acc ->
-      case fetch(map, key) do
-        {:ok, value} -> [{key, value} | acc]
-        :error -> acc
-      end
-    end)
-    |> :maps.from_list
+    do_take(keys, map, [])
+  end
```

Comment:
It is meant to take only lists AFAIK. Let's add a guard to take/drop/split here and on Keyword to make sure we enforce this API.